### PR TITLE
Update hevm to 0.51.3

### PR DESCRIPTION
**Pull request by arcz**
Closes https://github.com/crytic/echidna/issues/1085 and https://github.com/crytic/echidna/issues/1049.

### CODE REVIEW

File: lib/Echidna/Transaction.hs [hunk 1]
```diff
@@ -116,7 +116,7 @@ canShrinkTx Tx { call, gasprice = 0, value = 0, delay = (0, 0) } =
 canShrinkTx _ = True
 
 removeCallTx :: Tx -> Tx
-removeCallTx t = Tx NoCall 0 t.src 0 0 0 t.delay
+removeCallTx t = Tx NoCall t.src t.dst 0 0 0 t.delay
```
Comment (ggrieco-tob) [inline lib/Echidna/Transaction.hs:119]: Why this was changed?

Comment (arcz) [reply lib/Echidna/Transaction.hs:119]: `t.src` was in the wrong place and having 0 in the src caused a crash in tx generation which was previously not triggered as there was something deployed on the address 0. I think this was changed in hevm and now there is nothing deployed on 0.

**Comment by rappie** [comment]
Does this include the new cheatcodes for `diffusc`?

**Comment by arcz** [comment]
Not yet, unfortunately. It is still in progress https://github.com/ethereum/hevm/pull/259.

**Pull request merged**